ec/ecresource/projectresource/elasticsearch.go

```diff
@@ -0,0 +1,364 @@
+	updateBody := serverless.PatchElasticsearchProjectRequest{
+		Name: model.Name.ValueStringPointer(),
+	}
+
+	if model.Alias.ValueString() != "" {
+		updateBody.Alias = model.Alias.ValueStringPointer()
+	}
+
+	if util.IsKnown(model.SearchLake) {
+		updateBody.SearchLake = &serverless.OptionalElasticsearchSearchLake{}
+
+		if util.IsKnown(model.SearchLake.BoostWindow) {
+			boostWindow := int(model.SearchLake.BoostWindow.ValueInt64())
+			updateBody.SearchLake.BoostWindow = &boostWindow
+		}
+
+		if util.IsKnown(model.SearchLake.SearchPower) {
+			searchPower := int(model.SearchLake.SearchPower.ValueInt64())
+			updateBody.SearchLake.SearchPower = &searchPower
+		}
+	}
+
+	resp, err := es.client.PatchElasticsearchProjectWithResponse(ctx, model.Id.ValueString(), nil, updateBody)
+	if err != nil {
+		return diag.Diagnostics{
+			diag.NewErrorDiagnostic(err.Error(), err.Error()),
+		}
+	}
+
+	if resp.JSON200 == nil {
+		return diag.Diagnostics{
+			diag.NewErrorDiagnostic(
+				"Failed to update elasticsearch_project",
+				fmt.Sprintf("The API request failed with: %d %s\n%s",
+					resp.StatusCode(),
+					resp.Status(),
+					resp.Body),
+			),
+		}
+	}
+
+	return nil
+}
+
+func (es elasticsearchApi) EnsureInitialised(ctx context.Context, model resource_elasticsearch_project.ElasticsearchProjectModel) diag.Diagnostics {
+	id := model.Id.ValueString()
+	for {
+		resp, err := es.client.GetElasticsearchProjectStatusWithResponse(ctx, id)
+		if err != nil {
+			return diag.Diagnostics{
+				diag.NewErrorDiagnostic(err.Error(), err.Error()),
+			}
+		}
+
+		if resp.JSON200 == nil {
+			return diag.Diagnostics{
+				diag.NewErrorDiagnostic(
+					"Failed to get elasticsearch_project status",
+					fmt.Sprintf("The API request failed with: %d %s\n%s",
+						resp.StatusCode(),
+						resp.Status(),
+						resp.Body),
+				),
+			}
+		}
+
+		if resp.JSON200.Phase == serverless.Initialized {
+			return nil
+		}
+
+		es.sleeper.Sleep(200 * time.Millisecond)
+	}
+}
+
+func (es elasticsearchApi) Read(ctx context.Context, id string, model resource_elasticsearch_project.ElasticsearchProjectModel) (bool, resource_elasticsearch_project.ElasticsearchProjectModel, diag.Diagnostics) {
+	resp, err := es.client.GetElasticsearchProjectWithResponse(ctx, id)
+	if err != nil {
+		return false, model, diag.Diagnostics{
+			diag.NewErrorDiagnostic(err.Error(), err.Error()),
+		}
+	}
+
+	if resp.HTTPResponse != nil && resp.HTTPResponse.StatusCode == http.StatusNotFound {
+		return false, model, nil
+	}
+
+	if resp.JSON200 == nil {
+		return false, model, diag.Diagnostics{
+			diag.NewErrorDiagnostic(
+				"Failed to read elasticsearch_project",
+				fmt.Sprintf("The API request failed with: %d %s\n%s",
+					resp.StatusCode(),
+					resp.Status(),
+					resp.Body),
+			),
+		}
+	}
+
+	model.Id = basetypes.NewStringValue(id)
+	model.Alias = basetypes.NewStringValue(reformatAlias(resp.JSON200.Alias, id))
```

Comment:
I guess `reformatAlias` doesn't affect `alias` in `Update`, does it?

Reply:
tl;dr No, both Create and Update *must* use the user supplied `alias` property. 

There was some internal discussion about how `alias` is handled by the API, but the end result was that we'll keep the current behaviour for now. 

The API will append `-{short project id}` to whatever alias is specified in the API call, but will return the fully formed alias. So we must consistently send through the user supplied alias without the project id suffix, which means stripping it from the alias returned from the API.